test/validation/api/stash/stash.c

```diff
@@ -0,0 +1,1005 @@
+	CU_ASSERT(odp_stash_capability(&capa, ODP_STASH_TYPE_DEFAULT) == 0);
+	CU_ASSERT(capa.max_stashes_any_type > 0);
+	CU_ASSERT(capa.max_stashes > 0);
+	CU_ASSERT(capa.max_num_obj > 0);
+	CU_ASSERT(capa.max_obj_size >= sizeof(uint32_t));
+
+	memset(&capa, 0, sizeof(odp_stash_capability_t));
+	ret = odp_stash_capability(&capa, ODP_STASH_TYPE_FIFO);
+	CU_ASSERT(ret == 0);
+	CU_ASSERT(capa.max_stashes_any_type > 0);
+	if (ret == 0 && capa.max_stashes) {
+		CU_ASSERT(capa.max_num_obj > 0);
+		CU_ASSERT(capa.max_obj_size >= sizeof(uint32_t));
+	}
+}
+
+static void stash_param_defaults(void)
+{
+	odp_stash_param_t param;
+
+	memset(&param, 0xff, sizeof(odp_stash_param_t));
+	odp_stash_param_init(&param);
+	CU_ASSERT(param.type == ODP_STASH_TYPE_DEFAULT);
+	CU_ASSERT(param.put_mode == ODP_STASH_OP_MT);
+	CU_ASSERT(param.get_mode == ODP_STASH_OP_MT);
+	CU_ASSERT(param.cache_size == 0);
+}
+
+static void stash_create_u64(void)
+{
+	odp_stash_t stash, lookup;
+	odp_stash_param_t param;
+	uint32_t num = global.num_default.u64;
+
+	odp_stash_param_init(&param);
+	param.num_obj  = num;
+	param.obj_size = sizeof(uint64_t);
+
+	stash = odp_stash_create("test_stash_u64", &param);
+
+	CU_ASSERT_FATAL(stash != ODP_STASH_INVALID);
+
+	printf("\n    Stash handle: 0x%" PRIx64 "\n", odp_stash_to_u64(stash));
+
+	lookup = odp_stash_lookup("test_stash_u64");
+	CU_ASSERT(lookup != ODP_STASH_INVALID);
+	CU_ASSERT(stash == lookup);
+	CU_ASSERT(odp_stash_lookup("foo") == ODP_STASH_INVALID);
+
+	CU_ASSERT_FATAL(odp_stash_destroy(stash) == 0);
+}
+
+static void stash_create_u32(void)
+{
+	odp_stash_t stash, lookup;
+	odp_stash_param_t param;
+	uint32_t num = global.num_default.u32;
+
+	odp_stash_param_init(&param);
+	param.num_obj  = num;
+	param.obj_size = sizeof(uint32_t);
+
+	stash = odp_stash_create("test_stash_u32", &param);
+
+	CU_ASSERT_FATAL(stash != ODP_STASH_INVALID);
+
+	printf("\n    Stash handle: 0x%" PRIx64 "\n", odp_stash_to_u64(stash));
+
+	lookup = odp_stash_lookup("test_stash_u32");
+	CU_ASSERT(lookup != ODP_STASH_INVALID);
+	CU_ASSERT(stash == lookup);
+	CU_ASSERT(odp_stash_lookup("foo") == ODP_STASH_INVALID);
+
+	CU_ASSERT_FATAL(odp_stash_destroy(stash) == 0);
+}
+
+static void stash_create_u64_all(void)
+{
+	odp_stash_param_t param;
+	uint64_t input, output;
+	uint32_t i;
+	uint32_t num_obj = global.num_default.u32;
+	uint32_t num_stash = global.capa_default.max_stashes;
+	odp_stash_t stash[num_stash];
+
+	CU_ASSERT_FATAL(num_stash > 0);
+
+	odp_stash_param_init(&param);
+	param.num_obj  = num_obj;
+	param.obj_size = sizeof(uint64_t);
+
+	for (i = 0; i < num_stash; i++) {
+		stash[i] = odp_stash_create("test_stash_u64_all", &param);
+		CU_ASSERT_FATAL(stash[i] != ODP_STASH_INVALID);
+		CU_ASSERT_FATAL(odp_stash_get(stash[i], &output, 1) == 0);
+
+		input = i;
+		CU_ASSERT(odp_stash_put(stash[i], &input, 1) == 1);
+	}
+
+	for (i = 0; i < num_stash; i++) {
+		/* 1 msec delay to allow HW latency from put() to get() */
+		odp_time_wait_ns(ODP_TIME_MSEC_IN_NS);
```

Comment:
These (four) test cases merely test that stash creation succeeded and put only one item to each. I was thinking to keep those simple, and just move the 1msec delay before the loop and increase it e.g. to 100msec. Is that OK?

Reply:
It will work but may send a mixed message. I suggest doing:
for (j = 0; j < TIMEOUT; j++) {
  odp_time_wait_ns(ODP_TIME_MSEC_IN_NS);
  odp_stash_get(...);
}
CU_ASSERT(j != TIMEOUT);

Reply:
Fixed in v10.